iOS/UI/Swipeable Input/SwipeableInputAccessoryView.swift

```diff
@@ -394,6 +397,21 @@ class SwipeableInputAccessoryView: BaseView, UIGestureRecognizerDelegate, Active
                                 y: previewCenter.y + gravityVector.dy)
 
         previewView.center = previewCenter
+
+        // Provide haptic feedback when the message is ready to send.
+        let distanceToDropZone = CGVector(startPoint: previewCenter, endPoint: dropZoneCenter).magnitude
+        if distanceToDropZone < self.dropZoneFrame.height * 0.5 {
+            if !self.isPreviewInDropZone {
+                previewView.setBubbleColor(ThemeColor.D1.color, animated: true)
+                self.impactFeedback.impactOccurred()
+            }
+            self.isPreviewInDropZone = true
+        } else {
+            if self.isPreviewInDropZone {
+                previewView.setBubbleColor(ThemeColor.B1.color, animated: true)
+            }
+            self.isPreviewInDropZone = false
+        }
```

Comment:
Change the color and provide haptic feedback when the bubble moves into the drop zone.